src/charm.py

```diff
@@ -41,6 +41,18 @@
 SSL_CONFIG_FILE = "src/templates/ssl.conf.j2"
 
 
+class CheckFailed(Exception):
+    """Raise this exception if one of the checks in main fails."""
+
+    def __init__(self, msg, status_type=None):
+        """Raise this exception if one of the checks in main fails."""
+        super().__init__()
+
+        self.msg = msg
+        self.status_type = status_type
+        self.status = status_type(msg)
+
+
```

Comment:
Chisme to the rescue!  This can be replaced with:
```
from charmed_kubeflow_chisme.exceptions import ErrorWithStatus as CheckFailed
```

Reply:
actually, you already have it imported, and CheckFailed isn't used anyway.  I think this can be deleted?

Reply:
Yep. That's left overs. Removing.